### ISSUE TITLE
diagnostics with unconstrained parameters

### ISSUE THREAD

**Issue by bgoodri** [stan-dev/shinystan#65]
The various plots should have an checkbox to utilize the parameters in the unconstrained space. These can be obtained with something like

```
pars <- extract(stanfit, permuted = FALSE, inc_warmup = maybe)
skeleton <- get_inits(stanfit)[[1]]
upars <- apply(pars, 1:2, FUN = function(theta) {
  unconstrain_pars(stanfit, relist(theta, skeleton))
})
upars <- aperm(upars, c(2,3,1))
dimnames(upars) <- c(dimnames(pars)[1:2], list(parameters = 
                     head(dimnames(pars)[[3]], dim(upars)[3])))
```

**Comment by bob-carpenter** [comment]
I agree --- that's very useful because that's where the sampling's
actually happening.

- Bob

**Comment by jgabry** [comment]
I also agree. Right now in the unreleased development version there are text boxes that allow you to specify an arbitrary transformation by entering the body of an R function (e.g. to log transform you just need to type `log(x)` not `function(x) log(x)`). So you can tell it to transform to the unconstrained space if you know what the right transformation is. You can do this for the density, histogram, bivariate scatterplot, and trivariate 3D scatterplot on the **Explore** page. This will be included in v1.1.0 soon which I'm almost ready to tag and release. 

For v1.2.0 I can go the extra step and implement what you're suggesting here. I think it should still allow the user to enter whatever transformation they want, but also give the option to use the transformation to the unconstrained space that was used by Stan.
